pinot-plugins/pinot-input-format/pinot-parquet/src/main/java/org/apache/pinot/plugin/inputformat/parquet/ParquetRecordReader.java

```diff
@@ -32,17 +38,34 @@
  * It has two implementations: {@link ParquetAvroRecordReader} (Default) and {@link ParquetNativeRecordReader}.
  */
 public class ParquetRecordReader implements RecordReader {
+  private static final String AVRO_SCHEMA_METADATA_KEY = "parquet.avro.schema";
+
   private RecordReader _internalParquetRecordReader;
   private boolean _useAvroParquetRecordReader = true;
 
   @Override
   public void init(File dataFile, @Nullable Set<String> fieldsToRead, @Nullable RecordReaderConfig recordReaderConfig)
       throws IOException {
-    if (recordReaderConfig == null || ((ParquetRecordReaderConfig) recordReaderConfig).useParquetAvroRecordReader()) {
+    if (recordReaderConfig != null && ((ParquetRecordReaderConfig) recordReaderConfig).useParquetAvroRecordReader()) {
       _internalParquetRecordReader = new ParquetAvroRecordReader();
-    } else {
+    } else if (recordReaderConfig != null
+        && ((ParquetRecordReaderConfig) recordReaderConfig).useParquetNativeRecordReader()) {
       _useAvroParquetRecordReader = false;
       _internalParquetRecordReader = new ParquetNativeRecordReader();
+    } else {
+      // No reader type specified. Determine using file metadata
+      ParquetMetadata parquetMetadata =
+          ParquetFileReader.readFooter(new Configuration(), new Path(dataFile.getAbsolutePath()),
+              ParquetMetadataConverter.NO_FILTER);
+      Map<String, String> fileKeyValueMeta = parquetMetadata.getFileMetaData().getKeyValueMetaData();
```

Comment:
You can put this check `hasAvroSchemaInParquetFile()` inside `org.apache.pinot.plugin.inputformat.parquet.ParquetUtils` and reuse the same method inside `org.apache.pinot.plugin.inputformat.parquet.ParquetUtils.getParquetAvroSchema(...)`

Reply:
Ack